toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/WaiterThread.java

```diff
@@ -0,0 +1,67 @@
+package io.toast.tk.agent.ui;
+
+import java.io.IOException;
+
+import io.toast.tk.agent.config.AgentConfigProvider;
+import io.toast.tk.agent.run.TestRunner;
+
+public class WaiterThread implements Runnable {
+	  private WaiterPanel panel;
+	  private Thread thread;
+	  private TestRunner testrunner;
+	   
+	  public WaiterThread(AgentConfigProvider webConfigProvider) throws IOException { 
+		  	WaiterPanel waiterPanel = new WaiterPanel();
+			this.panel = waiterPanel;
+			this.testrunner = new TestRunner(webConfigProvider);   
+			this.thread = new Thread(new RunnerThread(testrunner));      
+	  }
+	  
+	  public void run() {
+		  	thread.start();
+	  		while(thread.isAlive()) {
+	  			try {
+					Thread.sleep(50);
+				} catch (InterruptedException e) {
+					Thread.currentThread().interrupt();
+				}
+	  			
+	  			if(panel.isInterupted()) {
+	  				thread.interrupt();
+	  			  	panel.setScript(testrunner.fileName, "Interupting");
+	  			  	testrunner.kill();
+	  			}
+	  			else {
+	  				panel.setScript(testrunner.fileName, "In progress");
+	  			}
+	  		}
+	  		panel.stop();
+	  		this.kill();
+	  }
+	  
+	  private void kill() {
+		  Thread.currentThread().interrupt();
+		  panel = null;
+		  thread.interrupt();
+		  thread = null;
+		  testrunner.kill();
+		  testrunner = null;
+	  }
+	  
+	  protected class RunnerThread implements Runnable {
+		  public TestRunner runner;
+		   
+		  public RunnerThread(TestRunner testrunner) throws IOException {           
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from constructor's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)